File: js/__tests__/loop_link_list.test.js
const LoopLinkList = require('../data_structures/loop_link_list')

describe('LoopLinkList', () => {

    let li;

    beforeEach(() => {
        li = new LoopLinkList()
    })

    it('show list', () => {
        li.insert('hello');
        li.insert('world');
        expect(li.toArr()).toEqual(['world', 'hello'])
    })

    it('move方法测试', () => {
        li.insert('hello');
        li.insert('world')
        li.insert('bear')
        li.insert('xiong', 'world')
        expect(li.toArr()).toEqual(['bear', 'world', 'xiong', 'hello'])
        li.move(2)
        expect(li.show()).toEqual('world')
        li.moveToHead()
        li.remove('xiong')
        li.move(3)
        expect(li.show()).toEqual('hello')
    })
})

File: py/__tests__/loop_linklist_test.py
import unittest
from data_structures.loop_linklist import LoopLinkList


class TestLoopLinkList(unittest.TestCase):
    def test_show(self):
        li = LoopLinkList()
        li.insert('hello')
        li.insert('bear')
        li.insert('xiong', 'hello')
        self.assertEqual(li.to_arr(), ['bear', 'hello', 'xiong'])

    def test_move(self):
        li = LoopLinkList()
        li.insert('leoo')
        li.insert('bear')
        li.insert('xx')
        self.assertEqual(li.to_arr(), ['xx', 'bear', 'leoo'])
        li.move(2)
        self.assertEqual(li.cur_node.item, 'bear')
        li.move_to_head()
        self.assertEqual(li.cur_node.item, 'head')
        li.move(4)
        self.assertEqual(li.cur_node.item, 'xx')
